tmt/test_checks/dmesg.py

```diff
@@ -0,0 +1,109 @@
+import datetime
+from typing import TYPE_CHECKING, List, Optional
+
+import tmt.log
+import tmt.steps.execute
+import tmt.steps.provision
+import tmt.utils
+from tmt.result import TestCheckResult
+from tmt.test_checks import TestCheckPlugin
+from tmt.utils import Path
+
+if TYPE_CHECKING:
+    from tmt.base import TestCheck
+
+
+TEST_POST_DMESG_FILENAME = 'tmt-dmesg.txt'
+
+
+@TestCheckPlugin.provides_check('dmesg')
+class DmesgTestCheck(TestCheckPlugin):
+    @classmethod
+    def _fetch_dmesg(
+            cls,
+            guest: tmt.steps.provision.Guest,
+            logger: tmt.log.Logger) -> tmt.utils.CommandOutput:
+
+        def _test_output_logger(
+                key: str,
+                value: Optional[str] = None,
+                color: Optional[str] = None,
+                shift: int = 2,
+                level: int = 3,
+                topic: Optional[tmt.log.Topic] = None) -> None:
+            logger.verbose(
+                key=key,
+                value=value,
+                color=color,
+                shift=shift,
+                level=level,
+                topic=topic)
+
+        return guest.execute(tmt.utils.ShellScript('dmesg'), log=_test_output_logger)
```

Comment:
Some error checking would be nice here maybe?
Or how will it behave if the command fails?

```
❯ podman run -it fedora dmesg
[root@2c0a7aa2d5cc /]# dmesg 
dmesg: read kernel buffer failed: Permission denied
```

Reply:
+1, and I think there's no test yet!